src/components/Search/Search.tsx

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
+import {TeamsType, UserDataType} from 'types/types';
+
+interface SearchProps<ObjectType> {
+    originalObject: ObjectType[];
+    updateFilteredObject: (updatedObject: ObjectType[]) => void;
+    notifyError: (hasError: boolean) => void;
+    searchProps: string[];
+    placeholder: string;
+}
+
+const Search = <ObjectType extends TeamsType | UserDataType>(props: SearchProps<ObjectType>): JSX.Element => {
+    const {originalObject, updateFilteredObject, notifyError, searchProps, placeholder} = props;
+
+    const [searchValue, setValue] = React.useState<string>('');
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => setValue(event.target.value);
+
+    React.useEffect(() => {
+        if (searchValue !== '') {
+            const filteredObject = originalObject.filter((obj) => {
+                let includes = false;
+                searchProps.forEach((prop) => {
+                    if (obj[prop] && obj[prop].toLowerCase().includes(searchValue)) {
+                        includes = true;
```

Comment:
for each prop, check if it exists in the object, and check for the searchValue on them